client/src/driver/BitBakeProjectScanner.ts

```diff
@@ -412,41 +412,81 @@ You should adjust your docker volumes to use the same URIs as those present on y
   }
 
   private async scanForRecipesPath (): Promise<void> {
-    const tmpFiles = this.searchFiles(this._recipesFileExtension)
+    const showRecipeFileNameOutput = await this.executeBitBakeCommand('bitbake-layers show-recipes -f')
+    if (showRecipeFileNameOutput.status !== 0) {
+      logger.error('Failed to scan recipes path')
+      return
```

Comment:
scanForRecipes() tends to throw rather than return on errors. There are other occurences in this patch.

Reply:
I make it throw like others.